### ISSUE TITLE
Port `barmpy` to PyMC

### ISSUE THREAD

**Issue by dvbuntu** [dvbuntu/barmpy#4]
[PyMC](https://www.pymc.io/welcome.html) is a python library that tries to fit bayesian models with Markov Chain Monte Carlo.  BARN essentially fits that mold, so it would be instructive and potentially useful to port `barmpy` to that ecosystem.  PyMC is a different approach from `sklearn`, however, so there may be a bit of learning curve.  Some good first steps:

1. Understand [PyMC-BART](https://www.pymc.io/projects/bart/en/latest/)
2. Port `BARN` to PyMC, using PyMC-BART as a starting point
3. Extract only needed MCMC components from PyMC to be used in `BARN`, keeping `sklearn` compatibility.

**Comment by dvbuntu** [comment]
Related but possibly distinct would be using `PyMC` or another backend to handle all the MCMC stepping.  This requires having sufficient control over the process to do the neural network proposal, training, and acceptance.  But it may simplify future development and take advantage of available improvements.  It was previously important to implement and fully understand myself, but as `barmpy` matures, it's likely more valuable to use the best possible tools where possible.
